src/services/impl/FileLogger.hh

```diff
@@ -21,7 +21,18 @@ class FileLogger : public ra::services::ILogger
 public:
     explicit FileLogger(const ra::services::IFileSystem& pFileSystem) noexcept
     {
-        m_pWriter = pFileSystem.AppendTextFile(pFileSystem.BaseDirectory() + L"RACache\\RALog.txt");
+        std::wstring sLogFilePath = pFileSystem.BaseDirectory() + L"RACache\\RALog.txt";
+
+        // if the file is over 1MB, rename it and start a new one
+        int64_t nLogSize = pFileSystem.GetFileSize(sLogFilePath);
```

Comment:
It's up to you but this variable can be destroyed earlier, either by scope or selection statement.

Reply:
Good point